lib/src/domain/usecase/implementation/genres_usecase_impl.dart

```diff
@@ -1,16 +1,44 @@
+import 'package:connectivity_plus/connectivity_plus.dart';
+
+import '../../../core/util/categories.dart';
 import '../../../core/util/states.dart';
-import '../../../data/datasource/remote/i_api_service.dart';
+import '../../../data/datasource/local/genre_db.dart';
 import '../../../data/repository/genre_repository_class.dart';
+import '../../model/genre.dart';
 import '../i_usecase.dart';
 
 class GenresUseCase implements IUseCase {
   final GenreRepository repository;
+  final Connectivity _connectivity = Connectivity();
+  final GenreDatabase genreDatabase;
 
-  GenresUseCase(this.repository, {IApiService? apiService});
+  GenresUseCase({
+    required this.repository,
+    required this.genreDatabase,
```

Comment:
usecase shouldn't know the database directly, instead use a local repository as a intermediary

Reply:
solved